File: leetcode/523-continuous-subarray-sum/Solution.java
import java.util.*;

class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {
        int sum = 0;
        Map<Integer, Integer> ct = new HashMap<>();
        ct.put(0, 0);
        for (int i = 0; i < nums.length; i++) {
            sum += nums[i];
            int mod = k != 0 ? sum%k : sum;
            if (!ct.containsKey(mod)) 
                ct.put(mod, i + 1);
            else if (i + 1 - ct.get(mod) >= 2)
                return true;
        }
        return false;
    }
}

File: leetcode/523-continuous-subarray-sum/Solution.py
from collections import Counter

class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        sums = [0]*(len(nums)+1)
        d = {0: 0}
        for i in range(0, len(nums)):
            sums[i+1] = sums[i] + nums[i]
            mod = sums[i+1] % k if k != 0 else sums[i+1]
            if mod in d:
                if i + 1 - d[mod] >= 2: return True
            else:
                d[mod] = i + 1
        return False
    